src/order/category/NonemptyFinLinOrd.lean

```diff
@@ -0,0 +1,82 @@
+/-
+Copyright (c) 2020 Johan Commelin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johan Commelin
+-/
+
+import data.fintype.sort
+import data.fin
+import order.category.LinearOrder
+
+/-! # Nonempty finite linear orders
+
+Nonempty finite linear orders form the index category for simplicial objects.
+-/
+
+universe variables u v
+
+open category_theory
+
+set_option old_structure_cmd true
+
+section prio
+
+-- see note [default priority]
+set_option default_priority 100
+
+/-- A typeclass for nonempty finite linear orders. -/
+class nonempty_fin_lin_ord (α : Type*) extends fintype α, decidable_linear_order α, order_bot α, order_top α.
+
+end prio
+
+instance punit.nonempty_fin_lin_ord : nonempty_fin_lin_ord punit :=
+begin
+  refine_struct
+  { .. punit.decidable_linear_ordered_cancel_add_comm_monoid,
+    .. punit.fintype };
+  { intros, exact punit.star <|> exact dec_trivial }
```

Comment:
Another `;`, `<|>` combo?

Reply:
@semorrison I'm not sure what you would like to see here. I think this is a perfect way of saying, hey... this will give you 10 goals, but they are all really easy: either `punit.star` or decidably trivial.

Reply:
Oh, I guess I didn't realise there were so many goals. No problem.